### PR TITLE
Check dashboards index pattern

### LINKED ISSUE #13300: [Metricbeat] Redis keys dashboard references non-existing index
Importing and opening the [[Metricbeat Redis] Keys ECS dashboard](https://github.com/elastic/beats/blob/master/metricbeat/module/redis/_meta/kibana/7/dashboard/Metricbeat-redis-keys.json) loads visualizations which point to non-existing indexes. Because of the missing index references, opening the dashboard in kibana shows broken vizualizations:

Screenshot 2019-08-21 at 12 37 58

- Version: 7.3.1
- Steps to Reproduce: 
1. `./metricbeat setup -E "setup.dashboards.url=https://staging.elastic.co/7.3.1-89dbfb4a/downloads/beats/beats-dashboards/beats-dashboards-7.3.1.zip"`
2. Open the "[Metricbeat Redis] Keys ECS" dashboard

### PR DESCRIPTION
**Pull request by jsoriano**
If a dashboard is created using an index pattern manually created,
it can contain incorrect references, check for index patterns to avoid
this mistake.

Fix #13300

### CODE REVIEW

**Comment by jsoriano** [comment]
jenkins, test this again

File: dev-tools/mage/check.go [hunk 1]
```diff
@@ -301,3 +318,19 @@ func checkTitle(re *regexp.Regexp, title string, module string) error {
...
+
+func checkDashboardIndexPattern(expectedIndex string, o *dashboardObject) error {
+	if objectMeta := o.Attributes.KibanaSavedObjectMeta; objectMeta != nil {
+		if index := objectMeta.SearchSourceJSON.Index; index != "" && index != expectedIndex {
+			return errors.Errorf("unexpected index pattern reference found in object meta: %s", index)
+		}
+	}
+	if visState := o.Attributes.VisState; visState != nil {
+		for _, control := range visState.Params.Controls {
+			if index := control.IndexPattern; index != "" && index != expectedIndex {
```
Comment (kaiyan-sheng) [inline dev-tools/mage/check.go]: When index will be ""?

Comment (jsoriano) [reply dev-tools/mage/check.go]: The field is optional, when it is not set it would be `""` here.

Comment (kaiyan-sheng) [reply dev-tools/mage/check.go]: oh I didn't know it's optional 😬 Thanks!

**Comment by jsoriano** [comment]
I found another place where wrong index patterns can appear, and we had a couple of dashboards affected in Filebeat, @kaiyan-sheng or @sayden could you take another look?

**Comment by jsoriano** [comment]
@kaiyan-sheng thanks for testing!

**Pull request merged**